e2e/tests/negative/node_reboot.robot

```diff
@@ -463,3 +463,392 @@ Power Off Replica Node Should Not Rebuild New Replica On Same Node
+    ...                Replica Node Level Soft Anti-Affinity is still the default value which is false
+    ...                Replica Zone Level Soft Anti-Affinity is still the default value which is true
+    ...                Create a testvol of 3 replicas and attach it to node-1
+    ...                Turn off node-2
+    ...                Wait for 6 minutes, turn on the node-2
+    ...                The volume should reuse the failed replica to rebuild. 
+    ...                Volume should become healthy again
+    ...
+    ...                Note: Test cases with replica-zone-soft-anti-affinity=false are skipped
+    ...                because the volume cannot be healthy in a single-zone cluster.
+    Given Set setting replica-replenishment-wait-interval to 300
+    And Set setting replica-disk-soft-anti-affinity to false
+    And Set setting replica-soft-anti-affinity to false
+    And Set setting replica-zone-soft-anti-affinity to true
+    And Create volume 0 with    dataEngine=${DATA_ENGINE}
+    And Attach volume 0 to node 0
+    And Record volume 0 replica names
+
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        When Power off node 1 for 6 mins
+        And Wait for longhorn ready
+        And Wait for volume 0 healthy
+
+        Then Check volume 0 replica names are as recorded
+    END
+
+
+Power Off Replica Node For More Than 5 Mins Should Reuse Failed Replica After Reboot Case 2
+    [Tags]  replica  reboot  anti-affinity  longhorn-10251  
+    [Documentation]    Anti-affinity Combination 2:
+    ...                Replica Disk Level Soft Anti-Affinity: false
+    ...                Replica Node Level Soft Anti-Affinity: true
+    ...                Replica Zone Level Soft Anti-Affinity: true
+    Given Set setting replica-replenishment-wait-interval to 300
+    And Set setting replica-disk-soft-anti-affinity to false
+    And Set setting replica-soft-anti-affinity to true
+    And Set setting replica-zone-soft-anti-affinity to true
+    And Create volume 0 with    dataEngine=${DATA_ENGINE}
+    And Attach volume 0 to node 0
+    And Record volume 0 replica names
+
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        When Power off node 1 for 6 mins
+        And Wait for longhorn ready
+        And Wait for volume 0 healthy
+    END
+
+
+Power Off Replica Node For More Than 5 Mins Should Reuse Failed Replica After Reboot Case 3
+    [Tags]  replica  reboot  anti-affinity  longhorn-10251
+    [Documentation]    Anti-affinity Combination 3:
+    ...                Replica Disk Level Soft Anti-Affinity: true
+    ...                Replica Node Level Soft Anti-Affinity: false
+    ...                Replica Zone Level Soft Anti-Affinity: true
+    Given Set setting replica-replenishment-wait-interval to 300
+    And Set setting replica-disk-soft-anti-affinity to true
+    And Set setting replica-soft-anti-affinity to false
+    And Set setting replica-zone-soft-anti-affinity to true
```

Comment:
The tests seem to be identical, consider using [test templates](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#test-templates) to make them data-driven.

Reply:
👍 cc @longhorn/qa @longhorn/dev 👆

Reply:
BTW, I suggest making the test case self-explanatory. Using general terms like adding a suffix `-case [1-n]` might make the final report harder to read and understand without first reviewing the test cases in detail.

Reply:
Current progress:
- Enhanced execution time improvements (saved one hour).
- Using test templates to make them data-driven.
- Still considering what naming convention for test cases would work best.

![Image](https://github.com/user-attachments/assets/02fe15c3-d126-48ea-b4f2-0f16e64e00b3)